html/docs/design/sync-ot.html

```diff
@@ -0,0 +1,435 @@
+              Taskwarrior users to consult and update the same task list on
+              multiple devices, without requiring constant connection.
+            </p>
+
+            <p>
+              The RFC proposes a new approach to synchronization that is not compatible
+              with the existing sync mechanism.  It has the following advantages:
+              <ul>
+                <li>Privacy-preserving: server does not have access to user data.</li>
+                <li>Format-agnostic: server does not know the format of the data, and therefore can be upgraded independently of replicas.</li>
+                <li>Performant: both server and replica performance scale linearly with the size of the replicas being synchronized.</li>
+                </ul>
+                A proof-of-concept of this model has been implemented in <a href="https://github.com/taskchampion/taskchampion">Taskchampion</a>a.
+            </p>
+
+            <p>
+              The remainder of this document summarizes the new model, and how it will be implemented in Taskwarrior.
+            </p>
+
+            <a name="model"></a>
+            <h3>Sync Model</h3>
+
+			<p>
+              Synchronization occurs between disconnected replicas, mediated by a server.
+			  The replicas never communicate directly with one another.
+			  The server does not have access to the task data; it sees only opaque blobs of data with a small amount of metadata.
+            </p>
+
+            <p>
+              The synchronization process is a critical part of the task database's functionality, and it cannot function efficiently without occasional synchronization operations.
+            </p>
+
+            <h4 id="operational-transforms">Operational Transforms</h4>
+
+            <p>
+            Synchronization is based on <a href="https://en.wikipedia.org/wiki/Operational_transformation">operational transformation</a>.
+            This section will assume some familiarity with the concept.
+            </p>
+
+            <h4 id="state-and-operations">State and Operations</h4>
+
+            <p>
+            At a given time, the set of tasks in a replica's storage is the essential <em>state</em> of that replica.
+            This state has the form of an unordered set of tasks indexed by UUID.
+            Each task is a collection of key/value pairs, both represented as strings.
+            <em>Any</em> state of this form is valid from the perspective of the synchronization model.
+            </p>
+
+            <p>
+            All modifications to that state occur via <em>operations</em>.
+            We can draw a network, or graph, with the nodes representing states and the edges representing operations.
+            For example:
+            </p>
+<pre>  o -- State: {"abc-123": {"description": "get groceries", "priority": "L"}}
+  |
+  | -- Operation: set abc-123 "priority" to "H"
+  |
+  o -- State: {"abc-123": {"description": "get groceries", "priority: "H"}
```

Comment:
```suggestion
  o -- State: {"abc-123": {"description": "get groceries", "priority: "H"}}
```

Reply:
@djmitche This one is still outstanding in the current revision.

Reply:
Oh, the "Commit Suggestion" button didn't work, oops.